catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/BroadcastBrickMessageUpdateTest.kt

```diff
@@ -0,0 +1,187 @@
+            message, firstBroadcastBrick,
+            baseActivityTestRule.activity
+        )
+        val spinnerValues = listOf(
+            getResourcesString(R.string.new_option),
+            getResourcesString(R.string.edit_option),
+            defaultMessage,
+            message
+        )
+        checkAllBrickSpinnerValues(spinnerValues)
+    }
+
+    @Test
+    fun testAllBroadcastBrickSpinnersContainTheEditedMessage() {
+        BroadcastMessageBrickTestUtils.editBroadcastMessageOnBrick(
+            defaultMessage, editedMessage, firstBroadcastBrick,
+            baseActivityTestRule.activity
+        )
+        val spinnerValues = listOf(
+            getResourcesString(R.string.new_option),
+            getResourcesString(R.string.edit_option),
+            editedMessage
+        )
+        checkAllBrickSpinnerValues(spinnerValues)
+    }
+
+    @Test
+    fun testAllBroadcastBrickSpinnersShowTheEditedMessage() {
+        BroadcastMessageBrickTestUtils.editBroadcastMessageOnBrick(
+            defaultMessage, editedMessage, firstBroadcastBrick,
+            baseActivityTestRule.activity
+        )
+        checkShowsCorrectSpinnerMessage(editedMessage)
+    }
+
+    @Test
+    fun testEditingOccursOnlyInCurrentScene() {
+        BroadcastMessageBrickTestUtils.editBroadcastMessageOnBrick(
+            defaultMessage, editedMessage, firstBroadcastBrick,
+            baseActivityTestRule.activity
+        )
+        switchScene()
+        checkShowsCorrectSpinnerMessage(defaultMessage)
+    }
+
+    private fun checkAllBrickSpinnerValues(spinnerValues: List<String>) {
+        onBrickAtPosition(1)
+            .onSpinner(R.id.brick_broadcast_spinner)
+            .checkNameableValuesAvailable(spinnerValues)
+        onBrickAtPosition(2)
+            .onSpinner(R.id.brick_broadcast_spinner)
+            .checkNameableValuesAvailable(spinnerValues)
+        onBrickAtPosition(3)
+            .onSpinner(R.id.brick_broadcast_spinner)
+            .checkNameableValuesAvailable(spinnerValues)
+        onBrickAtPosition(4)
+            .onSpinner(R.id.brick_broadcast_spinner)
+            .checkNameableValuesAvailable(spinnerValues)
+        onBrickAtPosition(5)
+            .onSpinner(R.id.brick_broadcast_spinner)
+            .checkNameableValuesAvailable(spinnerValues)
+    }
+
+    private fun checkShowsCorrectSpinnerMessage(message: String) {
+        onBrickAtPosition(1).onSpinner(R.id.brick_broadcast_spinner).checkShowsText(message)
+        onBrickAtPosition(2).onSpinner(R.id.brick_broadcast_spinner).checkShowsText(message)
+        onBrickAtPosition(3).onSpinner(R.id.brick_broadcast_spinner).checkShowsText(message)
+        onBrickAtPosition(4).onSpinner(R.id.brick_broadcast_spinner).checkShowsText(message)
+        onBrickAtPosition(5).onSpinner(R.id.brick_broadcast_spinner).checkShowsText(message)
+    }
+
+    private fun switchScene() {
+        baseActivityTestRule.finishActivity()
```

Comment:
Since you now don't call this function anywhere else:
I think the last activity is never finished this way, or is it?

Would that possibly cause any problems?

Reply:
I don't think so, because the finishActivity() should be called after @After by the JUnitRunner. 
This also means, that the calling of finishActivity() is not really needed. In all the teardowns.

Reply:
But I will check if there is a better way to restart an activity.

Reply:
In this case it is ok to use it like that.